core/frontend/src/render/webgl/Texture.ts

```diff
@@ -549,7 +549,9 @@ export class ExternalTextureLoader { /* currently exported for tests only */
     try {
       const texBytesPromise = req.imodel.getTextureImage({ name: req.name });
       texBytesPromise.then((texBytes: Uint8Array | undefined) => {
-        if (undefined !== texBytes && !req.imodel.isClosed) {
+        if (undefined === texBytes || req.imodel.isClosed) {
+          throw new Error("Requested a texture element that does not exist or iModel is closed.");

```

Comment:
iModel being closed should not produce an exception.

Reply:
It's also weird to `throw` here given you are inside a try-catch...does the backend not produce exceptions for nonexistent texture - it just returns `undefined` if any error occurs?

Reply:
It will throw an error if the name property isn't a string; otherwise, yes, it returns undefined if the texture cannot be retrieved.

Reply:
So make  your test pass something that's not a string to get an actual exception?

Reply:
I have the frontend ExternalTextureRequest interface set as string... ExternalLoadTexture.loadTexture() also explicitly takes a string

Reply:
This is Javascript. You can bypass the compiler's type checks.

Reply:
Fair enough - will do.

Reply:
Pushed a test that exercises the exception throwing (non-string name specified) and proved it works.